lib/beaker-pe/install/pe_utils.rb

```diff
@@ -327,11 +327,13 @@ def deploy_frictionless_to_master(host)
               node_group['classes'] ||= {}
             end
 
-            # add the pe_repo platform class
-            node_group['classes'][klass] = {}
+            # add the pe_repo platform class if it's not already present
+            if ! node_group['classes'].include?(klass)
+              node_group['classes'][klass] = {}
 
-            dispatcher.create_new_node_group_model(node_group)
-            on master, puppet("agent -t"), :acceptable_exit_codes => [0,2]
+              dispatcher.create_new_node_group_model(node_group)
+              on master, puppet("agent -t"), :acceptable_exit_codes => [0,2]
```

Comment:
I'm not sure I want to assume that work will *always* be performed. If it should do some work and doesn't, then agent install should fail later. If nothing fails later, then it must have worked alright.